src/sql/workbench/contrib/notebook/browser/notebook.component.ts

```diff
@@ -343,7 +342,16 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 
 	public unselectActiveCell() {
 		this.model.updateActiveCell(undefined);
-		this.detectChanges();
+	}
+
+	public updateActiveCell(cell: ICellModel) {
+		if (this._model.activeCell !== cell) {
+			this._model.updateActiveCell(cell);
+		}
+	}
+
+	public getNotebookCellLabel(cell: ICellModel): string {
+		return localize('notebookCelllabel', "{0} {1}", cell.cellType === CellTypes.Code ? 'Code Cell' : 'Markdown Cell', cell.id);
```

Comment:
You should be localizing "Code Cell" and "Markdown Cell" - just having the localize string be {0} {1} isn't actually doing anything.

But also - the cells themselves should have their labels. This shouldn't be the job of the Notebook component. This should just call cell.label or something.